plugin/mutate/README.md

```diff
@@ -1,78 +1,105 @@
-# Admin and other fun stuff
+# Dextool Mutate
 
-To get the files in the database:
-```sh
-sqlite3 dextool_mutate.sqlite3 "select * from files"
-```
+The Dextool mutation testing plugin. 
 
-The different states a mutant are found in Mutation.Kind.
+## Overview
 
-Reset all mutations of a kind to unknown which forces them to be tested again:
-```sh
-sqlite3 dextool_mutate.sqlite3 "update mutation SET status=0 WHERE mutation.kind=FOO"
-```
+Dextool's plugin for mutation testing of C/C++ projects. It can help you design new tests and evaluate the quality of existing  tests.
+
+### Features
+
+TODO: List supported mutation operators etc.
+TODO: What are the selling points etc?
 
-# Mutation Testing of Google Test
+# Getting Started
 
-This is an example of how to mutation test google test itself.
-It assumes the current directory is _build_ which is then located the google test repo.
+If you are new to dextool, we suggest you to read the following:
 
-Create a database of all mutation points:
+* TODO: Something about compilation databases
+* TODO: Something about mutation testing
+
+## Mutation testing
+
+### Using CMake
+
+The [Google Test project](https://github.com/google/googletest) will be used as an example.
+
+When setting up Dextool Mutate to analyze a CMake project, the typical workflow is to:
 ```sh
-cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
-dextool mutate analyze --compile-db compile_commands.json --restrict .. -- -D_POSIX_PATH_MAX=1024
+# Obtain the project
+git clone https://github.com/google/googletest.git
+cd googletest
+
+# Create a directory to hold the build output:
+mkdir build
+cd build
+
+# Generate a JSON compilation database and build scripts:
+cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -Dgtest_build_tests=ON -Dgmock_build_tests=ON ..
+make
 ```
 
-Reconfigure and prebuild with the tests activated:
+Suppose the source code is in ... and the test code is elsewhere. Use the ``--restrict`` option to specify what to analyze:
 ```sh
-cmake -Dgtest_build_tests=ON -Dgmock_build_tests=ON ..
-make
+dextool mutate analyze --compile-db compile_commands.json --restrict .. -- -D_POSIX_PATH_MAX=1024
 ```
 
-Create a file tester.sh with this content:
+Create a script ``tester.sh`` that runs the entire test suite when invoked:
```

Comment:
maybe add what I mentioned to you. That it is important that compiler.sh build all code, both test and program/SUT. it is important that test.sh only execute the tests.
Otherwise two things happens:
- many mutants will be wrongly classified as timeout when it was just the compilation that change a lot. GoogleTest is interesting in this case. Change a template and the compilation time may change to 1-2 minutes + 20s execution time of tests. Change a *hidden* implementation (.cpp) that only affects one translation unit and the compilation time is around a couple of seconds + 20s execution time.
- the mutation tester is unable to classify a mutation as "killed by compiler". it would be classified as "killed by the test suite". this results in a higher mutation score than it actually is. This is because the mutation score (for now) is calculated as survived / (killed+timeout+survived). "killed by compiler" is NOT part of the total because those are just invalid constructs. The programmer is never able to even write such code which goes against the assumption of a competent programmer that try to implement according to a specification.